common/src/main/java/org/terracotta/angela/common/util/IpUtils.java

```diff
@@ -31,18 +32,22 @@
     }
   }
 
-  public static boolean isLocal(String targetServerName) {
-    if (targetServerName.equals(LOCAL_HOSTNAME)) {
-      return true;
-    }
-
-    InetAddress address;
-    try {
-      address = InetAddress.getByName(targetServerName);
-    } catch (UnknownHostException e) {
-      throw new RuntimeException(e);
+  /**
+   * Try to determine (but not accurately) if a given name matches a local address.
+   * A local address is a local IP or local hostname.
+   * <p>
+   * This "name" can be any Angela generated names (i.e. in case of client arrays instanceId or symbolic names),
+   * we do not use "getByName()" to avoid any dns resolution
+   */
+  public static boolean isLocal(String name) {
+    switch (name) {
+      case "localhost":
+      case "127.0.0.1":
+      case " ::1":
+        return true;
+      default:
+        return LOCAL_HOSTNAME.equals(name) || name.startsWith("169.254.") || name.startsWith("fe80:");
```

Comment:
I think these checks are enough and provide more speed (7% gain on this project by removing unecessary dns resolution).
Most of the time, users specifying "local" names (**if they really need to**) would pick known addresses like the one listed here...

This method is mainly used to do some optimizations in angela to avoid for example an ssh remote access locally or un-necessary kit  downloads. So I think these checks are enough.